### PR TITLE
Fix/Enh: correctly handle ioloop timeouts with AsyncTestCase

### PR DESCRIPTION
**Pull request by gst**

### CODE REVIEW

**Comment by vitaly-krugl** [comment]
LGTM! Thank you.

**Comment by gst** [comment]
rebased for #683 merge conflict now resolved.

**Comment by vitaly-krugl** [comment]
@gst, could you please merge/rebase against master to resolve the merge conflict? Thx

**Comment by gst** [comment]
there, not 200% sure about how I fixed the merge conflict.. can you double-check it ?

File: tests/acceptance/async_test_base.py [hunk 1]
```diff
@@ -92,10 +94,12 @@ def on_open_error(self, connection, error):
 
     def on_timeout(self):
         """called when stuck waiting for connection to close"""
-        self.logger.info('on_timeout called at %s', datetime.utcnow())
-        # Initiate cleanup
-        self.timeout = None  # the dispatcher should have removed it
```
Comment (vitaly-krugl) [inline tests/acceptance/async_test_base.py]: @gst: this merge went bad. We need this line before `self.stop()`:

`self.timeout = None  # the dispatcher should have removed it`

**Comment by vitaly-krugl** [comment]
@gst, regarding 

> there, not 200% sure about how I fixed the merge conflict.. can you double-check it ? 

I do it like this: merge with the new master, then rebase against the new master.

File: tests/acceptance/async_test_base.py [hunk 2]
```diff
@@ -92,10 +94,12 @@ def on_open_error(self, connection, error):
...
+        self.logger.info('on_timeout at %s', datetime.utcnow())
```
Comment (vitaly-krugl) [inline tests/acceptance/async_test_base.py]: Let's have a single error log message that gives us useful information for debugging; e.g.,

```
        self.logger.error('%s timed out; on_timeout called at %s', 
            self, datetime.utcnow())
```

and get rid of `self.logger.error('on_timeout')` below

**Comment by gst** [comment]
here we go

**Comment by vitaly-krugl** [comment]
LGTM - thanks

**Pull request merged**